app/utils/validators/forms/AutoCompleteFieldBinding.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils.validators.forms
+
+import play.api.data.FormBinding
+import play.api.mvc.Request
+
+trait AutoCompleteFieldBinding {
+
+  // Suffix added by 'auto-complete-fix.js' to the form-binding key of the auto-complete input field.
+  private val suffix = "-autocomp"
+  private val suffixLen = suffix.length
+
+  def formValuesFromRequest(key: String)(implicit request: Request[_], formBinding: FormBinding): Map[String, Seq[String]] = {
```

Comment:
I found this in this comment where (I think) this implicit is coming from
```
object FormBinding {
  object Implicits {

    /**
     * Convenience implicit for testing and other scenarios where it's not important
     * to verify payload limits (e.g. prevent OutOfMemoryError's).
     *
     * Prefer using a FormBinding provided by PlayBodyParsers#formBinding since that honours play.http.parser.maxMemoryBuffer limits.
     */
    implicit val formBinding: FormBinding = new DefaultFormBinding(Form.FromJsonMaxChars, Form.FromJsonMaxDepth)
  }
}
```

Reply:
I think that's probably fine in this instance though